newIDE/app/src/Leaderboard/LeaderboardProvider.js

```diff
@@ -72,7 +73,9 @@ const reducer = (state: ReducerState, action: ReducerAction): ReducerState => {
       );
       const newCurrentLeaderboard = shouldDefineCurrentLeaderboardIfNoneSelected
         ? primaryLeaderboard || leaderboards[0]
-        : state.currentLeaderboard;
+        : (state.currentLeaderboard &&
+            leaderboardsByIds[state.currentLeaderboard.id]) ||
+          state.currentLeaderboard;
```

Comment:
hmm, I'm not sure how this logic will behave and why we need it?

Reply:
I'll try to make this logic clearer